magpie/job/magpie-job-hive-testbench

```diff
@@ -0,0 +1,147 @@
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  General Public License for more details.
+#  
+#  You should have received a copy of the GNU General Public License
+#  along with Magpie.  If not, see <http://www.gnu.org/licenses/>.
+#############################################################################
+
+# This script is the hive-testbench running script.  For the most part,
+# it shouldn't be editted.  See job submission files for configuration
+# details.
+
+source ${MAGPIE_SCRIPTS_HOME}/magpie/lib/magpie-lib-hadoop-helper
+source ${MAGPIE_SCRIPTS_HOME}/magpie/lib/magpie-lib-log
+source ${MAGPIE_SCRIPTS_HOME}/magpie/lib/magpie-lib-paths
+
+export PATH=${PATH}:${HIVE_HOME}/bin:${HADOOP_HOME}/bin
+
+if [ "${HIVE_TESTBENCH_TYPE}" != "tpch" ] && [ "${HIVE_TESTBENCH_TYPE}" != "tpcds" ]
+then
+    echo "Invalid testbench type!  Exiting."
+    exit 1
+fi
+
+if [ "${HIVE_CLI_VERSION}" != "beeline" ] && [ "${HIVE_CLI_VERSION}" != "hive" ]
+then
+    echo "Invalid Hive CLI!  Exiting."
+    exit 1
+fi
+
+if [ "${HIVE_TESTBENCH_DIR}X" == "X" ]
+then
+    testbenchdir=""
+else
+    testbenchdir="${HIVE_TESTBENCH_DIR}"
+fi
+
+if [ ! -d ${HIVE_TESTBENCH_DIR} ]
+then
+    echo "Testbench path is not a directory"
+    exit 1
+fi
+
+if [ "${HIVE_TESTBENCH_TYPE}X" == "X" ]
+then
+    testbenchtype="tpch"    
+else
+    testbenchtype="${HIVE_TESTBENCH_TYPE}"
+fi
+
+if [ "${HIVE_TESTBENCH_DATA_SIZE}X" == "X" ]
+then
+    # if not specified for some reason, default to 20GB
+    testbenchsize=20
+else
+    testbenchsize=$HIVE_TESTBENCH_DATA_SIZE
+fi
+
+if [ "${HIVE_TESTBENCH_QUERY_COUNT}X" == "X" ]
+then
+    testbenchquerycount=16
+else
+    testbenchquerycount=$HIVE_TESTBENCH_QUERY_COUNT
+fi
+
+for node in `cat ${HIVE_CONF_DIR}/masters`
+do
+
+    echo "*******************************************************"
+    echo "* Executing Hive Testbench data generation"
+    echo "*******************************************************"
+
+    cd $testbenchdir
+    command="${testbenchdir}/${testbenchtype}-setup.sh ${testbenchsize}"
+    echo "Attempting to execute ${command} on `hostname`"
+    $command
+
+    cmdpid=$!
+    wait $cmdpid
+
+    echo "*******************************************************"
+    echo "* Submitting Hive Testbench queries"
+    echo "*******************************************************"
+    
+    # set the database names and query file prefixes based on testbench type
+    if [ "${testbenchtype}" == "tpch" ]
+    then
+        testbenchdb="${testbenchtype}_flat_orc_${testbenchsize}"
+        queryfileprefix="tpch_query"
+    else
+        testbenchdb="${testbenchtype}_bin_partitioned_orc_${testbenchsize}"
+        queryfileprefix="query"
+    fi
+    
+    # set the CLI prefix based on the configuration
+    if [ "${HIVE_CLI_VERSION}" == "beeline" ]
+    then
+        cliprefix="${HIVE_HOME}/bin/beeline -u jdbc:hive2://localhost:10000/${testbenchdb}"
```

Comment:
I don't know the functionality perfectly, but this doesn't seem right.  Shouldn't it be HIVE_MASTER_NODE instead of localhost?  And shouldn't it be some defined port number from somewhere else, and not just hard coded to 10000?

Reply:
You are correct...I've gone through and added/updated for HIVE_MASTER_NODE and HIVE_PORT.